.changeset/tasty-fireants-switch.md

```diff
@@ -0,0 +1,5 @@
+---
+"@fuel-ts/abi-typegen": patch
+---
+
+fix: make Predicate configurables option in typegen
```

Comment:
```suggestion
fix: make configurables properties optional in typegen
```

Reply:
@nedsalk Thanks for your suggestion. 

I am having a hard time understanding how this issue extends to Contracts and Scripts. 

Could you please provide a detailed outline of the error and an example of a scenario where this issue occurs for Contracts and Scripts?

Reply:
For contracts, you would have to change [this line](https://github.com/FuelLabs/fuels-ts/blob/8c5171d947291814c6bca0334d72379e51167167/packages/abi-typegen/src/templates/contract/dts.hbs#L44) to include the `Partial<...>` as you did for the predicates. The relevant script file to change is [this one](https://github.com/FuelLabs/fuels-ts/blob/8c5171d947291814c6bca0334d72379e51167167/packages/abi-typegen/src/templates/script/factory.hbs#L47). Upon changing it, you'll find that some tests are failing and you'll have to do a change like you did [here](https://github.com/FuelLabs/fuels-ts/pull/2815/files#diff-50b9975b1c89e23c847d731fb5c02edaac999b537242e2056e29451e904cae3fR19).

Reply:
@nedsalk Ok so for Scripts and Contracts this only results in error if you use the generated configurable type.

That is why I did not understand the request.

The error is flagged for predicates as soon as you try to instantiate the predicate with a partial configurable.

But for Scripts and Contracts it only happens if you type safe the configurable object:

```ts
const configurable: ConfigurableContractAbiConfigurables {
   ...
}
```

Reply:
Okay, so configurables for contracts and scripts aren't taken into account in the generated factory code. This is one more reason to go for #2686 which will make it easy to add this type safety.